install/0000_30_machine-api-operator_04_metal3provisioning.crd.yaml

```diff
@@ -81,6 +81,24 @@ spec:
               description: provisioningOSDownloadURL is the location from which the OS Image used to boot
                 baremetal host machines can be downloaded by the metal3 cluster.
               type: string
+            provisioningNetwork:
```

Comment:
This new field seems confusing to me.

It seems like it should be
'provisioningNetworkDisabled' and it should default to false.

It seems like the state of manage or unmanaged is mostly superfluous.  It doesn't seem there's an intended case of ProvisioningDHCPExternal and provisioningNetworkManaged being true.

Could you provide a matrix of expected options around this field correlating with other fields?

Reply:
The enhancement request discussing this is here (also linked above): https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/baremetal-provisioning-optional.md

ProvisioningDHCPExternal is deprecated.

Reply:
ProvisioningDHCPExternal allows us to have 2 modes , "External" or not.  We currently need to manage 3 states with the Provisioning Network and hence have moved to adding a new field for that. ProvisioningDHCPExternal is in the deprecation path and in 4.6 we will continue to support ProvisioningDHCPExternal.